### ISSUE TITLE
`lines-around-comment`: add `allowFuncStart` / `allowFuncEnd` configs

### ISSUE THREAD

**Issue by c3-epopham** [eslint-stylistic/eslint-stylistic#745]
### Clear and concise description of the problem

I want to be able to configure parameters for allowing functions to start/end with comments that are not pre-prepended by a newline character so that they do not conflict with a `prettier` rule about new lines.

Currently, the `@typescript-eslint/lines-around-comment` rule can conflict with `prettier` in scenarios like this:

```ts
return concat(
  // ACTION_DESCRIPTION_1
  of(ACTION_1),

  // ACTION_DESCRIPTION_2
  of(ACTION_2),
```

With this code, `@typescript-eslint/lines-around-comment` throws an error on the `ACTION_DESCRIPTION_1` line for `Expected line before comment`.

If I add a newline before this comment, `prettier` throws an error on the newline for `Delete <newline_character>`.

There is not currently a way to configure these rules to work together nicely.

### Suggested solution

Extend (current rule)[https://github.com/prettier/eslint-config-prettier?tab=readme-ov-file#lines-around-comment-deprecated] to have configurations for allowing functions to start/end with a comment line.
```json
{
  "rules": {
    "lines-around-comment": [
      "error",
      {
        "beforeBlockComment": true,
        "afterBlockComment": true,
        "beforeLineComment": true,
        "afterLineComment": true,
        "allowBlockStart": true,
        "allowBlockEnd": true,
        "allowObjectStart": true,
        "allowObjectEnd": true,
        "allowArrayStart": true,
        "allowArrayEnd": true,
        // start solution
        "allowFuncStart": true,
        "allowFuncEnd": false
        // end solution
      }
    ]
  }
}
```

### Alternative

I am able to use `// eslint-disable-next-line` to avoid this error being triggered, but I think there should be a way to avoid leveraging this disable flag for this scenario.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by c3-epopham** [comment]
I think this is similar to Issue https://github.com/eslint-stylistic/eslint-stylistic/issues/462
